pkg/infra/usagestats/service/service.go

```diff
@@ -82,22 +85,27 @@ func (uss *UsageStats) Run(ctx context.Context) error {
 	for {
 		select {
 		case <-sendReportTicker.C:
-			if traceID, err := uss.sendUsageStats(ctx); err != nil {
-				uss.log.Warn("Failed to send usage stats", "error", err, "traceID", traceID)
-			}
-
-			lastSent = time.Now()
-			if err := uss.kvStore.Set(ctx, "last_sent", lastSent.Format(time.RFC3339)); err != nil {
-				uss.log.Warn("Failed to update last sent time", "error", err)
+			if uss.readyToReport {
```

Comment:
Might be cleaner to just revert the changes below and use something like this?
```suggestion
			if !uss.readyToReport {
				nextSendInterval = time.Minute
				sendReportTicker.Reset(nextSendInterval)
				continue
			}
```